src/cswinrt/strings/WinRT.cs

```diff
@@ -448,31 +491,65 @@ public static unsafe (ObjectReference<IActivationFactoryVftbl> obj, int hr) GetA
         {
             Marshal.ThrowExceptionForHR(Platform.CoDecrementMTAUsage(_mtaCookie));
         }
-    }
-
+    }
+
     internal class BaseActivationFactory
     {
-        private readonly ObjectReference<IActivationFactoryVftbl> _IActivationFactory;
+        private Tuple<ObjectReference<IActivationFactoryVftbl>, IntPtr> _IActivationFactory;
```

Comment:
This should still be marked as `volatile`, no? You're reading it here concurrently, but writing with an interlocked op.